_posts/2024-06-26-devfile-ai-assistant.adoc

```diff
@@ -0,0 +1,334 @@
+---
+title: Introducing the Devfile AI Assistant
+layout: post
+author: Anatolii Bazko
+description: >-
+  The Devfile AI Assistant is an online tool that simplifies the creation of devfiles for developers.
+categories: []
+keywords: ['devfile', 'ChatGPT', 'AI']
+slug: /@tolusha/devfile-ai-assistant
+---
+
+== Introducing the Devfile AI Assistant
+
+In software development, it's important to be efficient. Whether you're an experienced developer or just starting, setting up your development environment can be complex and time-consuming. That's where the link:https://chatgpt.com/g/g-Bm20CP2Rp-devfile-assistant[Devfile AI Assistant] comes in. It's designed to help developers create devfiles easily without needing in-depth knowledge of the devfile syntax or structure. It uses a link:https://devfile.io[devfile knowledge base] to help you craft a devfile, making sure your development environment is set up right.
+
+Another tool that simplifies creating devfile is the link:https://github.com/devfile/vscode-walkthrough-extension[VS Code Walkthrough Extension for Devfile]. It provides an interactive guide to help users understand and utilize devfiles within Visual Studio Code. Using a wizard, you can easily create a devfile to customize your Cloud Development Environment.
+
+
+== What is a devfile?
+
+Devfiles are configuration files that define development environments. They specify the components, commands, events etc. that make up a development workspace. They play a central role in the workflow of Eclipse Che.
+
+In Eclipse Che, devfiles are used to automate the setup of development environments. By providing a devfile, developers can ensure a consistent environment across the team, getting rid of the "works on my machine" problems. When a devfile is applied, Eclipse Che provisions the specified containers, clones the necessary repositories, and sets up the workspace with all the tools and dependencies required for development. It makes them an essential tool for modern cloud-based development.
+
+image::/assets/img/devfile-ai-assistant/devfile-eclipse-che-cde.png[Devfile->Eclipse-Che->CDE]
+{nbsp} +
+
+== Step-by-step example of how to create a devfile
+
+Let's check out some real-life examples of how the Devfile AI Assistant can help you in creating devfiles for your projects."
+
+. Start by providing a basic devfile.
++
+**Request:**
++
+[source]
+----
+Provide a basic devfile.
+----
+**Response:**
++
+[source,yaml]
+----
+schemaVersion: 2.3.0
+metadata:
+  name: basic-devfile
+components:
+  - name: tools
+    container:
+      image: quay.io/devfile/universal-developer-image:latest
+----
+
+. Add a project to the devfile.
++
+**Request:**
++
+[source]
+----
+Add https://github.com/eclipse-che/che-server git project.
```

Comment:
I agree with Mario, that project part is a bit controversial. I think we need to add a note about that

Reply:
Suggestion - https://github.com/eclipse-che/blog/pull/55/files#r1648590270